### ISSUE TITLE
PDF'ing using Jupyter

### ISSUE THREAD

**Issue by eddy-ojb** [jupyter/notebook#2458]
Hello,

Jupyter is a superb tool but is let down by it's pdf'ing capabilities. There are a number of us in the data science community that require the use of Jupyter to run our code and export outputs. Jupyter gives the illusion that the code is run on a paper page. It also allows markdown and html editing, latex and  export options. However, there is little point in doing any of this if one has to transpose the notes straight in to another editor like Word, for presenting documents.

Most of the world uses pdf'ing and we all print documents and scribble on them, at least from time to time. What I think would make Jupyter a lot more popular would be to allow proper pdf'ing capabilities. This would mean we could make professional grade documents, hide code inputs and present the outputs with, for example, commentary.

At the moment, I don't see Jupyter as anything more than a scratch pad for notes and outputting images in a data science context.

Are there plans to improve the presentation capabilities of Jupyter for e.g. pdf'ing?

Ed

**Comment by mpacer** [comment]
We are acutely aware of the importance of PDF (among other) export capabilities. 

Have you spent any time looking at https://github.com/jupyter/nbconvert? There has been a great deal of effort placed into converting to PDF, LaTeX, markdown, RST, AsciiDoc, and other output formats.

The next release of nbconvert has the ability to  hide input and ouptut cells globally. If you want an example of how to do it on a cell-by-cell basis please look at https://github.com/mpacer/hiding_tags_nbconvert/.  You can include commentary on your code as part of your markdown cells or in-line with comments. 

It is unclear what you mean beyond those examples of "proper pdf'ing capabilities". I also do not know what you mean by "none of it carries to a pdf document" — with the exception of raw `html` formatting  injected into markdown cells, markdown content should be reflected in the output pdf. 

Additionally, there are the `Download as…` options in the File menu of the notebook itself. 

Thank you for raising your concerns and as an open source project, we always welcome new contributions!

**Comment by eddy-ojb** [comment]
Hi mpacer,

Thank you for getting back to me. I've already looked at nbconvert but I don't know what I'm doing. I can't see any examples with a clear corresponding explanation as to what's going on. I've tried e.g. [](https://github.com/jupyter/nbconvert-examples/tree/master/pdf)

There isn't much context in this set of examples and when looking at the notebook, there is nothing that indicates that the associated .py script has been used.

I would be interested in blogging examples but my problem is that I have no idea what I'm doing without some basic examples at the minimum to get started.  The ability to show/hide output cells globally looks handy but only if the documents can be properly formatted.

In order to try and format a notebook, I would like to be able to set page margins, line wrapping, line breaks, caption editing etc. I have tried  using, for example, markdown code for line break and page break in the cells to achieve this, followed by e.g. the 'download as pdf' tool within Jupyter but the formatting doesn't carry to the pdf. I've uploaded an example notebook for reference in case I'm being super clear as to what I want to achieve:

[](https://github.com/eddy-ojb/How-to-Format-and-PDF-Jupyter-Notebooks)

Thanks

Ed

**Comment by JWock82** [comment]
I agree completely with eddy-ojb. Jupyter looks promising, but as a practicing structural engineer it's worthless if I can't create professional looking formatted pdf's of my calculations to hand to my non-programming peers for review. Many of my peers don't know or want to learn python. I need to be able to hide individual code cells, set margins and text-wrapping, as well as have it print to pdf the way it looks in my browser.
